src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -11527,6 +11538,9 @@ void sofia_handle_sip_i_invite(switch_core_session_t *session, nua_t *nua, sofia
 	}
 
 	channel = tech_pvt->channel = switch_core_session_get_channel(session);
+	if (switch_channel_var_true(channel, "handle_update") || sofia_test_pflag(profile, PFLAG_HANDLE_UPDATE)) {
```

Comment:
The check and validation should be the same as above, if handle_update channel var is set then it take precedence over the sofia profile flag.